frontend/www/js/omegaup/contest/edit.ts

```diff
@@ -110,6 +110,50 @@ OmegaUp.on('ready', () => {
           })
           .catch(ui.apiError);
       },
+      addCatToLanguages: (): void => {
+        const contest: types.ContestAdminDetails = {
+          admin: true,
+          admission_mode: 'private',
+          alias: contestEdit.details.alias,
+          archived: false,
+          available_languages: {},
+          director: '',
+          opened: false,
+          penalty_calc_policy: 'sum',
+          problemset_id: 0,
+          show_penalty: true,
+          title: contestEdit.details.title,
+          description: contestEdit.details.description,
+          has_submissions: contestEdit.details.has_submissions,
+          start_time: contestEdit.details.start_time,
+          finish_time: contestEdit.details.finish_time,
+          points_decay_factor: contestEdit.details.points_decay_factor,
+          submissions_gap: contestEdit.details.submissions_gap,
+          languages: contestEdit.details.languages,
+          feedback: contestEdit.details.feedback,
+          penalty: contestEdit.details.penalty,
+          scoreboard: contestEdit.details.scoreboard,
+          penalty_type: contestEdit.details.penalty_type,
+          show_scoreboard_after: contestEdit.details.show_scoreboard_after,
+          partial_score: contestEdit.details.partial_score,
+          needs_basic_information: contestEdit.details.needs_basic_information,
+          requests_user_information:
+            contestEdit.details.requests_user_information,
+          contest_for_teams: contestEdit.details.contest_for_teams,
+        };
+        contest.languages.push('cat');
+        api.Contest.update({
+          ...contest,
+          contest_alias: contest.alias,
+          alias: null,
+          teams_group_alias: contestEdit.teamsGroup?.alias,
+          contest_for_teams: !!contestEdit.teamsGroup?.alias,
```

Comment:
```suggestion
          contest_alias: contest.alias,
          languages: contestEdit.details.languages.concat(['cat']),
```

aquí no es necesario pasarle _todos_ los campos posibles. el API actualiza los campos que se le proporcionaron y deja todos los demás tal cual.